packages/waku/src/lib/handlers/dev-worker-impl.ts

```diff
@@ -158,7 +158,8 @@ const mergedViteConfig = await mergeUserViteConfig({
 });
 
 const vitePromise = createViteServer(mergedViteConfig).then(async (vite) => {
-  await vite.ws.close();
+  const HMRKEY = vite.hot ?? vite.ws;
```

Comment:
Great progress, this can be better.

```
const hot = viteHot(vite);
hot.close()
```

And viteHot would do the fallbacking. Same with other places.